Revamp and Expand Component Style Guide

zihanKuang opened the pull request:
This PR fix #14821

### Changes included

-   **Fix: Table of Contents Visibility**
    -   Resolved an issue where the side table of contents was not displaying correctly on the page, restoring proper navigation.

-   **Content Expansion and Renaming**
    -   Renamed the guide from "Component Shape Guide" to "Component Style Guide" to accurately reflect its new, broader scope.
    -   Expanded the content to include detailed explanations for Kubernetes components, integrated technologies, arrows, and connectors, in addition to the original generic shapes.

-   **Structural and UX Improvements**
    -   Component descriptions are now organized into logical categories (e.g., Foundational Shapes, Symbolic Shapes, Flowchart Shapes).
    -   Implemented collapsible sections (`<details>`) to present this information cleanly, preventing the page from becoming too long and making it easier for users to navigate.
    -   Added new visuals, including GIFs and static images, to effectively demonstrate component styles and their use within Kanvas.

leecalcote commented on docs/pages/extensions/kanvas/component-style-guide.md:
```diff
@@ -0,0 +1,183 @@
+display-title: "false"
+list: include
+type: extensions
+category: kanvas
```
needs changed.

l5io commented:
🚀 Preview for commit 2375b223b540ed0657714675912332607bc2baa8 at: https://684323be1ca7dbb5fc9ce692--meshery-docs-preview.netlify.app

l5io commented:
🚀 Preview for commit 9f1e1562ca11c113417908cf21e583653affd105 at: https://68440410490c3fa20df9d4ee--meshery-docs-preview.netlify.app

zihanKuang commented:
Hi @leecalcote 
Based on the suggestions from the meeting, I've made the following changes:

* Updated a concise page covering Meshery content and placed it under "Configuration Management."
* Added a detailed page to the Kanvas documentation: https://github.com/layer5io/docs/pull/598

leecalcote commented:
@vishalvivekm feedback?

l5io commented:
🚀 Preview for commit d23c9964a757ff5382a7a2e14d8cf1e7c0468233 at: https://68506003ccd94b28a59d4f9b--meshery-docs-preview.netlify.app
